python/src/aac/lang/_active_context.py

```diff
@@ -0,0 +1,165 @@
+"""The Active Context maintains the highly-contextual AaC DSL."""
+import copy
+from attr import Factory, attrib, attrs, validators
+
+from aac.parser import ParsedDefinition
+from aac.lang.definition_helpers import (
+    get_definition_by_name,
+    get_definitions_by_root_key,
+    search_definition,
+    remove_list_type_indicator,
+)
+
+
+@attrs(slots=True, auto_attribs=True)
+class ActiveContext:
```

Comment:
Should the ActiveContext manage the list of actively-contributing plugins and sources? This would allow us to control the application of plugins at the context level which will have implications for things like tests.